libraries/standard/mqtt/src/iot_mqtt_operation.c

```diff
@@ -540,31 +541,61 @@ static bool _sendPingRequest( _mqttConnection_t * pMqttConnection )
 
     pPingreqOperation = &( pMqttConnection->pingreq );
 
-    /* Because PINGREQ may be used to keep the MQTT connection alive, it is
-     * more important than other operations. Bypass the queue of jobs for
-     * operations by directly sending the PINGREQ in this job. */
-    bytesSent = pMqttConnection->pNetworkInterface->send( pMqttConnection->pNetworkConnection,
-                                                          pPingreqOperation->u.operation.pMqttPacket,
-                                                          pPingreqOperation->u.operation.packetSize );
+    /* Only send the PINGREQ if the keep-alive period has elapsed since the connection
```

Comment:
The logic of determining whether or not to send a PINGREQ can be moved to```_IotMqtt_ProcessKeepAlive``` for readability. 

Also, update the following comment in ```_IotMqtt_ProcessKeepAlive``` to indicate that reschedule the job to run on next keep alive interval if either a PINGREQ is sent or we determine that there is no need to send a PINGREQ because the connection is active. 

```
/* When a PINGREQ is successfully sent, reschedule this job to check for a
 * response shortly. */
```

Reply:
Done